### PR TITLE
extra-bundled-libraries: align naming convention with ghc

### PR DESCRIPTION
**Pull request by DavidEichmann**
---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [x] Any changes that could be relevant to users have been recorded in the changelog.
* [x] The documentation has been updated, if necessary.
* [x] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: Cabal/Distribution/Simple/BuildPaths.hs [hunk 1]
```diff
@@ -218,6 +220,24 @@ mkStaticLibName platform (CompilerId compilerFlavor compilerVersion) lib
   = "lib" ++ getHSLibraryName lib ++ "-" ++ comp <.> staticLibExtension platform
   where comp = prettyShow compilerFlavor ++ prettyShow compilerVersion
 
+-- | Create a library name for a bundled shared library from a given name.
+-- This matches the naming convention for shared libraries as implemented in
+-- GHC's packageHsLibs function in the Packages module.
+-- If the given name is prefixed with HS, then this prepends 'lib' and appends
+-- the compiler flavour/version and shared library extension e.g.:
+--     "HSrts-1.0" -> "HSrts-1.0-ghc8.7.20190109.so"
```
[the remark this answers is not in the document]
Comment (DavidEichmann) [reply Cabal/Distribution/Simple/BuildPaths.hs]: Absolutely right, will fix.

**Comment by alpmestan** [comment]
Some CI jobs fail with:

```
Distribution/Simple/BuildPaths.hs:61:1: Warning:
    The import of ‘isPrefixOf’ from module ‘Data.List’ is redundant
```

I'm not sure why we only see this on some of them (e.g https://travis-ci.org/haskell/cabal/jobs/483113957).

**Comment by 23Skidoo** [comment]
@hvr Is this good to go now?

**Comment by hvr** [comment]
yeah, much better :-)

**Pull request merged**